### ISSUE TITLE
Warning: file_put_contents(): Filename cannot be empty in C:\xampp\htdocs\vqmod\vqmod.php on line 245 MODS CACHE PATH NOT WRITEABLE

### ISSUE THREAD

**Issue by GoogleCodeExporter** [rohitrng/vqmod#84]
```
NOTE THAT THIS IS FOR VQMOD ENGINE ERRORS ONLY. FOR GENERAL ERRORS FROM
MODIFICATIONS CONTACT YOUR DEVELOPER

What steps will reproduce the problem?
1. install vqmod 2.3.1 on opencart (I'm testing on localhost)
2.
3.

What is the expected output? What do you see instead?

Expected to see my cart. 
I see this: Warning: file_put_contents(): Filename cannot be empty in 
C:\xampp\htdocs\vqmod\vqmod.php on line 245
MODS CACHE PATH NOT WRITEABLE

Installing a previous version of vqmod works fine.

vQmod Version: 3.2.1
Server Operating System: windows 7, xampp.


Please provide any additional information below.

```

Original issue reported on code.google.com by `Travis.L...@gmail.com` on 27 Feb 2013 at 4:16

**Comment by GoogleCodeExporter** [comment]
```
The error kinda speaks for itself. You need to make sure you have write 
permission for the mods.cache path (this is /vqmod/mods.cache )
```

Original comment by `DJG6...@gmail.com` on 27 Feb 2013 at 10:25
- Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
version 2.3.0 works perfectly, so it would seem that my permissions are all 
write (no pun intended). I tried upgrading to 2.3.1 from 2.3.0 and it wouldn't 
work. So I went and did a fresh install of opencart (1.5.5.1) and put 2.3.0 on 
it, worked fine. Un-installed opencart, reinstalled a fresh 1.5.5.1 then tried 
installing 2.3.1. 2.3.1 didn't work. So I deleted all 2.3.1 files from the 
server, then installed 2.3.0 on the same opencart installation, and it worked 
like a charm.

So in summary, 2.3.1 would not work where 2.3.0 would. All folders and files 
that needed write permissions had them because 2.3.0 worked. Hopefully it is 
just me, but vqmod 2.3.1 didn't work with opencart 1.5.5.1.
```

Original comment by `Travis.L...@gmail.com` on 27 Feb 2013 at 10:51

**Comment by GoogleCodeExporter** [comment]
```
Open /vqmod/vqmod.php
Find line 244
$modCache = $this->path($this->modCache);
Change it to
$modCache = $this->path($this->modCache, true);

That should fix it
```

Original comment by `DJG6...@gmail.com` on 27 Feb 2013 at 11:04

**Comment by GoogleCodeExporter** [comment]
Original comment by `DJG6...@gmail.com` on 27 Feb 2013 at 11:08
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Just released 2.3.2 to fix this bug
```

Original comment by `DJG6...@gmail.com` on 27 Feb 2013 at 11:13

**Comment by GoogleCodeExporter** [comment]
```
I have applied this change on ver1.5.4 but the error is still there as 
"MODS CACHE PATH NOT WRITEABLE". Please give some advise to resolve this error.

Thanks
```

Original comment by `jtndrpa...@gmail.com` on 3 May 2013 at 12:48

**Comment by GoogleCodeExporter** [comment]
```
As with the first comment, you need to give write permissions on the file and 
containing folder
```

Original comment by `DJG6...@gmail.com` on 3 May 2013 at 2:16
